File: Cribbage/AIs/Hand/RandomHandAI.cpp
#include "RandomHandAI.h"

#include <vector>
#include <algorithm>

#include "../../Utils/Card.h"
#include "../../Utils/Random.h"

std::vector<Card> RandomHandAI::select(const std::vector<Card>& dealt, bool isMyCrib)
{
    std::vector<Card> keep;

    while (keep.size() < 4)
    {
        int i = Random::Instance()->randInt() % 6;
        if (std::find(keep.begin(), keep.end(), dealt[i]) == keep.end())
        {
            keep.push_back(dealt[i]);
        }
    }

    return keep;
}

File: CribbageFlutter/cribbage/lib/models/ai/hand/random_hand_ai.dart
import 'dart:math';
import './hand_ai.dart';
import '../../deck/card.dart';

class RandomHandAI implements HandAI {
  final Random _rand = Random();

  List<Card> select(List<Card> dealt, bool isMyCrib) {
    List<Card> keep = [];

    while (keep.length < 4) {
      int i = _rand.nextInt(dealt.length);
      if (!keep.contains(dealt[i])) {
        keep.add(dealt[i]);
      }
    }

    return keep;
  }
}
